eng/pipelines/armstrong-validation.yml

```diff
@@ -0,0 +1,88 @@
+trigger: none
+
+jobs:
+- job:
+  pool:
+    name: azsdk-pool-mms-ubuntu-2204-general
+    vmImage: ubuntu-22.04
+  
+  # refer to https://github.com/MicrosoftDocs/pipelines-go/blob/main/azure-pipelines.yml
+  variables:
+    GOBIN:  '$(GOPATH)/bin' # Go binaries path
+    GOROOT: '/usr/local/go1.22.2' # Go installation path
+    GOPATH: '$(system.defaultWorkingDirectory)/gopath' # Go workspace path
+    modulePath: '$(GOPATH)/src/github.com/$(build.repository.name)' # Path to the module's code
+
+  steps:
+  - bash: |
+      ls -al $(Build.SourcesDirectory)
+      ls -al $(Build.SourcesDirectory)/eng
+      ls -al $(Build.SourcesDirectory)/eng/scripts
+    displayName: bash list eng/scripts
+
+  - pwsh: |
+      Get-ChildItem -Path $(Build.SourcesDirectory)
+      Get-ChildItem -Path $(Build.SourcesDirectory)/eng
+      Get-ChildItem -Path $(Build.SourcesDirectory)/eng/scripts
+    displayName: pwsh list eng/scripts
+
+  - task: GoTool@0
+    inputs:
+      version: '1.22.2'
+  
+  - bash: |
+      ls -al $(Build.SourcesDirectory)
+      ls -al $(Build.SourcesDirectory)/eng
+      ls -al $(Build.SourcesDirectory)/eng/scripts
+    displayName: bash list eng/scripts
+
+  - pwsh: |
+      Get-ChildItem -Path $(Build.SourcesDirectory)
+      Get-ChildItem -Path $(Build.SourcesDirectory)/eng
+      Get-ChildItem -Path $(Build.SourcesDirectory)/eng/scripts
+    displayName: pwsh list eng/scripts
+
+  - script: |
```

Comment:
I believe something in this script is deleting/moving too many files on the agent.  After this script, folder $(Build.SourcesDirectory) only contains folder `gopath`.

https://dev.azure.com/azure-sdk/playground/_build/results?buildId=3735358&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=9304a5f1-6fc9-5147-1787-f6d3d736a40b&l=15

Reply:
I'm also not sure if you need to do any of this, I would have expected the `GoTool` task to set everything up correctly.  You might be able to see what pipelines in repo https://github.com/Azure/azure-sdk-for-go are doing.

Reply:
Thanks for the debugging. I have resolved the issue